src/js/components/Image.js

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames/dedupe';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Util from '../utils/Util';
+
+const METHODS_TO_BIND = ['onImageError'];
+
+class Image extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      imageErrorCount: 0
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    let {src, fallbackSrc} = this.props;
+    let newSrc;
+    if (src !== nextProps.src) {
+      newSrc = nextProps.src;
+    }
+
+    if (src === nextProps.src && fallbackSrc !== nextProps.fallbackSrc) {
+      newSrc = nextProps.fallbackSrc;
+    }
+
+    // Try again if a new url was provided
+    if (newSrc) {
+      let image = ReactDOM.findDOMNode(this);
+      image.src = newSrc;
```

Comment:
What's the purpose of setting `src` directly on the node? Why not a more "React" approach by setting state?